acquire-driver-common/tests/integration/switch-storage-identifier.cpp

```diff
@@ -0,0 +1,242 @@
+    EXPECT(file_size >= 64 * 48 * nframes,
+           "Expected file to have size at least %d (has size %d): %s",
+           64 * 48 * nframes,
+           file_size,
+           file_path.c_str());
+}
+
+void
+configure_storage_raw(AcquireRuntime* runtime)
+{
+    CHECK(runtime);
+
+    AcquireProperties props = {};
+    OK(acquire_get_configuration(runtime, &props));
+
+    auto* dm = acquire_device_manager(runtime);
+    CHECK(dm);
+
+    props.video[0].storage = { 0 };
+    DEVOK(device_manager_select(dm,
+                                DeviceKind_Storage,
+                                SIZED("Raw"),
+                                &props.video[0].storage.identifier));
+
+    storage_properties_set_filename(&props.video[0].storage.settings,
+                                    SIZED(TEST ".bin") + 1);
+
+    OK(acquire_configure(runtime, &props));
+}
+
+void
+validate_storage_raw()
+{
+    const std::string file_path = TEST ".bin";
+    EXPECT(
+      fs::exists(file_path), "Expected file to exist: %s", file_path.c_str());
+    EXPECT(fs::is_regular_file(file_path),
+           "Expected file to exist: %s",
+           file_path.c_str());
+    const auto file_size = fs::file_size(file_path);
+    EXPECT(file_size == (sizeof(VideoFrame) + 64 * 48) * nframes,
+           "Expected file to have size %d (has size %d): %s",
+           64 * 48 * nframes,
+           file_size,
+           file_path.c_str());
+}
+
+void
+acquire(AcquireRuntime* runtime)
+{
+    CHECK(runtime);
+
+    OK(acquire_start(runtime));
+    OK(acquire_stop(runtime));
+}
+
+int
+main()
+{
+    auto* runtime = acquire_init(reporter);
+    try {
+        configure_camera(runtime);
+
+        configure_storage_trash(runtime);
+        acquire(runtime);
+
+        configure_storage_tiff(runtime);
+        acquire(runtime);
+        validate_storage_tiff();
+        fs::remove_all(TEST ".tif");
+
+        configure_storage_trash(runtime);
+        acquire(runtime);
+
+        configure_storage_raw(runtime);
+        acquire(runtime);
+        validate_storage_raw();
+        fs::remove_all(TEST ".bin");
+
+        configure_storage_trash(runtime);
+        acquire(runtime);
+
+        configure_storage_tiff(runtime);
+        acquire(runtime);
+        validate_storage_tiff();
+        fs::remove_all(TEST ".tif");
+
+        configure_storage_raw(runtime);
+        acquire(runtime);
+        validate_storage_raw();
+        fs::remove_all(TEST ".bin");
+
+        configure_storage_trash(runtime);
+        acquire(runtime);
+
+        configure_storage_raw(runtime);
+        acquire(runtime);
+        validate_storage_raw();
+        fs::remove_all(TEST ".bin");
+
+        configure_storage_tiff(runtime);
+        acquire(runtime);
+        validate_storage_tiff();
+        fs::remove_all(TEST ".tif");
+
+    } catch (const std::exception& exc) {
+        return 1;
+    } catch (...) {
+        return 1;
+    }
+
+    acquire_shutdown(runtime);
```

Comment:
```suggestion
    } catch (const std::exception& exc) {
        acquire_shutdown(runtime);
        return 1;
    } catch (...) {
        acquire_shutdown(runtime);
        return 1;
    }

    acquire_shutdown(runtime);
```

I recall that it's important to always try to shutdown, if possible.